delfin/drivers/huawei/oceanstor/oceanstor.py

```diff
@@ -178,6 +178,209 @@ def list_volumes(self, context):
             raise exception.StorageBackendException(
                 'Failed to get list volumes from OceanStor')
 
+    def list_controllers(self, context):
+        try:
+            # Get list of OceanStor controller details
+            controllers = self.client.get_all_controllers()
+
+            controller_list = []
+            for controller in controllers:
+                status = constants.ControllerStatus.NORMAL
+                if controller['RUNNINGSTATUS'] == consts.STATUS_CTRLR_UNKNOWN:
+                    status = constants.ControllerStatus.UNKNOWN
+                if controller['RUNNINGSTATUS'] == consts.STATUS_CTRLR_OFFLINE:
+                    status = constants.ControllerStatus.OFFLINE
+
+                c = {
+                    'name': controller['NAME'],
+                    'storage_id': self.storage_id,
+                    'native_controller_id': controller['ID'],
+                    'status': status,
+                    'location': controller['LOCATION'],
+                    'soft_version': controller['SOFTVER'],
+                    'cpu_info': controller['CPUINFO'],
+                    'memory_size': controller['MEMORYSIZE'],
+                }
+                controller_list.append(c)
+
+            return controller_list
+
+        except Exception as err:
+            LOG.error("Failed to get controller metrics from OceanStor: {}"
+                      .format(err))
+            raise exception.StorageBackendException(
```

Comment:
It would be better that not wrap all the exception to one specific exception.
Same as below.

Reply:
Thanks, done.